### PR TITLE
Domains: Add breadcrumbs to DNS records page placeholder

### PR DESCRIPTION
**Pull request by rafaelgallani**
#### Changes proposed in this Pull Request
Adds the breadcrumbs for the DNS records page when in the loading state.

#### Preview
image

#### Testing instructions
- Go to `/domains/manage/:domain/dns/:domain`;
- Check that when the page loads it looks like the screenshot in the preview section;

_You can also try setting the `showPlaceholder` prop for the `DnsRecords` component as `true`._

### CODE REVIEW

File: client/my-sites/domains/domain-management/dns/dns-records.jsx [hunk 1]
```diff
@@ -105,13 +112,21 @@ class DnsRecords extends Component {
 		);
 	}
 
+	renderPlaceholder() {
+		return config.isEnabled( 'domains/dns-records-redesign' ) ? (
+			<DomainMainPlaceholder breadcrumbs={ this.renderBreadcrumbs } />
+		) : (
+			<DomainMainPlaceholder goBack={ this.goBack } />
+		);
+	}
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-records.jsx:121]: I believe this component will only be rendered when the feature flag is enabled, as the feature flag check is at the controller level (`domain-management/controller.jsx`). So I think you can just return `<DomainMainPlaceholder breadcrumbs={ this.renderBreadcrumbs } />` here.

Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-records.jsx]: I believe this component will only be rendered when the feature flag is enabled, as the feature flag check is at the controller level (`domain-management/controller.jsx`). So I think you can just return `<DomainMainPlaceholder breadcrumbs={ this.renderBreadcrumbs } />` here.

File: client/my-sites/domains/domain-management/dns/add-dns-record.jsx [hunk 2]
```diff
@@ -33,6 +34,12 @@ class AddDnsRecprd extends Component {
 		selectedSite: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ).isRequired,
 	};
 
+	constructor( props ) {
+		super( props );
+		this.renderBreadcrumbs = this.renderBreadcrumbs.bind( this );
+		this.goBack = this.goBack.bind( this );
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/add-dns-record.jsx]: I wonder if we should use arrow functions to not need to bind methods 🤔

**Review by leonardost — APPROVED** [review]
Tested and LGTM! 👍 Left just some comments that are not blockers.

File: client/my-sites/domains/domain-management/dns/dns-records.jsx (hunk 1 above)
Comment (rafaelgallani) [reply client/my-sites/domains/domain-management/dns/dns-records.jsx]: Thanks for that. 
Confirmed that the changes didn't affect the page without the feature flag: 

image

**Comment by rafaelgallani** [comment]
It all looks good now, except there is a rebase pending and I also noticed some small changes between the different breadcrumbs across the pages:

image

image

I'll also add these changes to this PR.

**Comment by rafaelgallani** [comment]
☝🏻  Rebase on `trunk`

**Comment by rafaelgallani** [comment]
@leonardost can you take a final look at it?

I added the change to the breadcrumbs' style to this PR, but there's still a minor difference: 

It looks grayish on the nameservers page:

image

But on the records page, it looks white (since the page background is white):

image

I'll leave it as is for now, since this PR won't do any changes to the nameservers page - which I think will probably be removed in a future follow-up PR. Let me know if you find anything else 😄

**Comment by leonardost** [comment]
@rafaelgalani thanks for doing the updates! Everything LGTM 💯 
I don't think the background is a problem - as you said, the name servers page will eventually be removed, so it's ok 🙂

**Pull request merged**